File: ios/Classes/Models/HMSRoleExtension.swift
//
//  HMSRoleExtension.swift
//  hmssdk_flutter
//
//  Copyright © 2021 100ms. All rights reserved.
//

import Foundation
import HMSSDK


class HMSRoleExtension{
    
    static func toDictionary(role:HMSRole)-> Dictionary<String,Any?>{
        
        let dict:[String:Any?] = [
            "name":role.name,
            "publish_settings":HMSPublishSettingExtension.toDictionary(publishSettings:role.publishSettings),
            "subscribe_settings":HMSSubscribeSettingsExtension.toDictionary(subscribeSettings:role.subscribeSettings),
            "priority":role.priority,
            "permissions":HMSPermissionExtension.toDictionary(permission:role.permissions),
            "general_permissions":role.generalPermissions,
            "internal_plugins":role.internalPlugins,
            "external_plugins":role.externalPlugins,
        ]
        
        return dict
    }
    
    
}

File: android/src/main/kotlin/live/hms/hmssdk_flutter/HMSRoleExtension.kt
package live.hms.hmssdk_flutter

import android.annotation.SuppressLint
import live.hms.hmssdk_flutter.hms_role_components.PermissionParamsExtension
import live.hms.hmssdk_flutter.hms_role_components.PublishParamsExtension
import live.hms.hmssdk_flutter.hms_role_components.SubscribeSettings
import live.hms.video.sdk.models.role.HMSRole

class HMSRoleExtension {
    companion object {
        @SuppressLint("LongLogTag")
        fun toDictionary(role: HMSRole?): HashMap<String, Any?>? {
            val hashMap = HashMap<String, Any?>()
            if (role == null)return null

            hashMap["name"] = role?.name ?: "unknown"
            hashMap["publish_settings"] = PublishParamsExtension.toDictionary(role?.publishParams ?: null)
            hashMap["subscribe_settings"] = SubscribeSettings.toDictionary(role?.subscribeParams ?: null)
            hashMap["priority"] = role?.priority!!
            hashMap["permissions"] = PermissionParamsExtension.toDictionary(role.permission ?: null)

            return hashMap
        }
    }
}
